app/components/mandatarissen/update-state.js

```diff
@@ -134,11 +142,33 @@ export default class MandatarissenUpdateState extends Component {
       start: this.date,
       status: this.newStatus,
       einde: endDate,
-    });
+    };
+
+    const newMandataris = this.store.createRecord(
+      'mandataris',
+      newMandatarisProps
+    );
+
+    if (this.selectedReplacement) {
+      const replacementMandataris =
+        await this.mandatarisService.getOrCreateReplacement(
+          this.args.mandataris,
+          this.selectedReplacement,
+          // passing these along because if we pass the model, relations will be
+          // evaluated as of right now and we haven't saved yet
+          newMandatarisProps,
+          this.selectedFractie
+        );
+      newMandataris.tijdelijkeVervangingen = [replacementMandataris];
```

Comment:
I'm a bit confused about this, it is possible to have multiple vervangingen, but we completely replace the array here?